nemo_skills/pipeline/openrlhf/ppo.py

```diff
@@ -0,0 +1,446 @@
+        )
+        return cmd
+
+    def get_common_rl_arg_overrides(self, cluster_config):
+        cmd = (
+            " --micro_rollout_batch_size 16 "
+            " --rollout_batch_size 1024 "
+            " --n_samples_per_prompt 1 "
+            " --actor_learning_rate 5e-7 "
+            " --critic_learning_rate 9e-6 "
+            " --init_kl_coef 0.01 "
+            " --normalize_reward "
+            " --vllm_sync_backend nccl "
+        )
+        return cmd
+
+    def format_wandb_args(self, cluster_config, disable_wandb, wandb_project, expname):
+        if not disable_wandb:
+            if os.getenv('WANDB_API_KEY') is None:
+                raise ValueError("WANDB_API_KEY is not set. Use --disable_wandb to disable wandb logging")
+
+            cmd = (f" --use_wandb $WANDB_API_KEY "
+                   f" --wandb_project {wandb_project} "
+                   f" --wandb_run_name {expname} ")
+        else:
+            cmd = ""
+
+        return cmd
+
+    def get_preamble_cmd(self, cluster_config):
+        cmd = " echo 'No preamble command to execute, skipping...' "
+        return cmd
+
+    def get_job_cmd(self, cluster_config):
+        ray_job_cmd = self.get_ray_launch_cmd(cluster_config)
+        ray_job_cmd = (
+            f"echo 'Starting training' && "
+            f"{ray_job_cmd} -m openrlhf.cli.train_ppo_ray "
+            f"  {self.format_reward_critic_args(cluster_config)} "
+            f"  {self.format_actor_args(cluster_config)} "
+            f"  {self.format_train_args(cluster_config)} "
+            f"  {self.format_data_args(cluster_config)} "
+            f"  {self.get_common_arg_overrides(cluster_config)} "
+            f"  {self.get_common_rl_arg_overrides(cluster_config)} "
+            f"  {self.logging_params} "
+            f"  {self.extra_arguments} "
+        )
+        return ray_job_cmd
+
+    def get_cmd(self, cluster_config):
+
+        self.logging_params = self.format_wandb_args(cluster_config, self.disable_wandb, self.wandb_project, self.expname)
+        preamble_cmd = self.get_preamble_cmd(cluster_config)
+
+        cmd = (
+            f"export HYDRA_FULL_ERROR=1 && "
+            f"export PYTHONPATH=$PYTHONPATH:/nemo_run/code && "
+            f"export TRITON_CACHE_DIR=/nemo_run/code/.triton_cache && "
+            f"cd /nemo_run/code && "
+            f"echo 'Running preamble command: '{preamble_cmd}' && "
+            f"{preamble_cmd} && "
+        )
+
+        ray_job_cmd = self.get_job_cmd(cluster_config)
+        ray_server_cmd = self.get_ray_server_cmd(ray_job_cmd, cluster_config)
+
+        cmd = f"{cmd} {ray_server_cmd} "
+        return cmd
+
+
+def get_training_cmd(
+    cluster_config,
+    task: Optional[PPOOpenRLHFTask],
+    partition,
+    hf_model,
+    rm_model,
+    output_dir,
+    prompt_data,
+    num_gpus,
+    num_nodes,
+    expname,
+    disable_wandb,
+    wandb_project,
+    extra_arguments,
+):
+    if 'timeouts' not in cluster_config:
```

Comment:
let's probably remove this and replace with a TODO as currently this is not used